Description: Fix build problem on mips*
  Make sure required defines are set on mips.
Author: YunQiang Su <wzssyqa@gmail.com>
Debian-Bug: https://bugs.debian.org/824190
Forwarded: no
Reviewed-By: Petter Reinholdtsen <pere@hungry.com>
Last-Update: 2016-05-25

Index: zfs-linux-0.6.5.6/lib/libspl/include/sys/isa_defs.h
===================================================================
--- zfs-linux-0.6.5.6.orig/lib/libspl/include/sys/isa_defs.h
+++ zfs-linux-0.6.5.6/lib/libspl/include/sys/isa_defs.h
@@ -172,7 +172,24 @@ extern "C" {
 #define	_BIG_ENDIAN
 #define	_SUNOS_VTOC_16
 
-#else /* Currently x86_64, i386, arm, powerpc, s390, and sparc are supported */
+/* MIPS arch specific defines */
+#elif defined(__mips__)
+
+#if defined(__MIPSEB__)
+#define	_BIG_ENDIAN
+#elif defined(__MIPSEL__)
+#define	_LITTLE_ENDIAN
+#else
+#error MIPS no endian specified
+#endif
+
+#ifndef _LP64
+#define _ILP32
+#endif
+
+#define	_SUNOS_VTOC_16
+
+#else /* Currently x86_64, i386, arm, powerpc, s390, sparc and mips are supported */
 #error "Unsupported ISA type"
 #endif
 
Index: zfs-linux-0.6.5.6/lib/libefi/rdwr_efi.c
===================================================================
--- zfs-linux-0.6.5.6.orig/lib/libefi/rdwr_efi.c
+++ zfs-linux-0.6.5.6/lib/libefi/rdwr_efi.c
@@ -88,7 +88,8 @@ struct dk_map2  default_vtoc_map[NDKMAP]
 #if defined(_SUNOS_VTOC_16)
 
 #if defined(i386) || defined(__amd64) || defined(__arm) || \
-    defined(__powerpc) || defined(__sparc) || defined(__s390__)
+    defined(__powerpc) || defined(__sparc) || defined(__s390__) || \
+    defined(__mips__)
 	{	V_BOOT,		V_UNMNT	},		/* i - 8 */
 	{	V_ALTSCTR,	0	},		/* j - 9 */
 
